caffeine@patapon.info/extension.js

```diff
@@ -182,27 +181,38 @@ class CaffeineToggle extends QuickSettings.QuickMenuToggle {
     _syncTimers(resetDefault) {
         this._itemsSection.removeAll();
         this._timerItems.clear();
-        const durationIndex = this._settings.get_int(DURATION_TIMER_INDEX);
-
-        for (const timer of TIMERS) {
+        const variantDuration = this._settings.get_value(DURATION_TIMER_LIST);
+        const durationValues = variantDuration.deepUnpack();
+        const shortTimer = durationValues[0];
+        const mediumTimer = durationValues[1];
+        const longTimer = durationValues[2];
+
+        // Create menu timer
+        for (const [index, timer] of [shortTimer, mediumTimer, longTimer, 0].entries()) {
```

Comment:
Can `shortTimer`, `mediumTimer` and `longTimer` be dropped, and instead use `[...durationValues, 0].entries()`?

I don't use the spread operator much, so I may be wrong, but this seems like a good opportunity for it

Reply:
Yes, I can remove these constants and simplify this part.